var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -295,6 +323,41 @@ def cmake_args(self):
                 '-DBZIP2_LIBRARIES=%s' % spec['bzip2'].libs.joined(';')
             ])
 
+        # Doxygen documentation
+        options.extend([
+            '-DDEAL_II_COMPONENT_DOCUMENTATION=%s' %
+            ('ON' if '+doc' in spec else 'OFF'),
+        ])
```

Comment:
The CMake package has a helper function that simplifies defines for both boolean and string variants:
```suggestion
        options.append(self.define_from_variant('doc', 'DEAL_II_COMPONENT_DOCUMENTATION'))
```

Reply:
That's great! I didn't know about this feature. I'll incorporate your suggestion, which is a great improvement on how things are currently done.

Reply:
I've taken a look at some other packages, and I think that the arguments to `define_from_variant` should be transposed. Is this correct?

Reply:
Yes sorry, my bad!

Reply:
No worries! I just wanted to double check that I understood the syntax. I appreciate the tip. I've managed to to a lot of simplification by using it.